src/main/java/org/mule/service/http/impl/service/client/HttpMessageLogger.java

```diff
@@ -31,17 +34,24 @@ public HttpMessageLogger(final LoggerType loggerType) {
 
   @Override
   public void onDataReceivedEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
   @Override
   public void onDataSentEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
-  private void logBuffer(Buffer buffer) {
+  private static Map<String, String> getLogContext(Connection connection) {
+    Map<String, String> logContext = (Map<String, String>) connection.getAttributes().getAttribute("logContext");
```

Comment:
I did a load test of 10k invocations with 50 concurrent clients and then analyzed the number of log lines per correlation id and the numbers are even so there doesn't seem to be a leak of the contexts across invocations